websubhub-ballerina/content_distribution.bal

```diff
@@ -0,0 +1,122 @@
+isolated function processPublishRequestAndRespond(http:Caller caller, http:Request request, http:Headers headers,
+                                                  map<string> params, HttpToWebsubhubAdaptor adaptor) returns error? {
+    http:Response response = new;
+    string? topic = getEncodedValueOrUpdatedErrorResponse(params, HUB_TOPIC, response);
+    if topic is () {
+        return error Error("Could not find the `hub.topic` parameter");
+    } else {
+        string contentType = request.getContentType();
+        UpdateMessage updateMsg = check createUpdateMessage(contentType, topic, request);
+        Acknowledgement|UpdateMessageError|error updateResult = adaptor.callOnUpdateMethod(updateMsg, headers);
+        response.statusCode = http:STATUS_OK;
+        if (updateResult is Acknowledgement) {
+            response.setTextPayload("hub.mode=accepted");
+            response.setHeader("Content-type","application/x-www-form-urlencoded");
+        } else if (updateResult is UpdateMessageError) {
+            var errorDetails = updateResult.detail();
+            updateErrorResponse(response, errorDetails["body"], errorDetails["headers"], updateResult.message());
+        } else {
+            var errorDetails = UPDATE_MESSAGE_ERROR.detail();
+            updateErrorResponse(response, errorDetails["body"], errorDetails["headers"], updateResult.message());
+        }
+        respondToRequest(caller, response);
+    }
+}
+
+# Creates the `websubhub:UpdateMessage` from the `http:Request`.
+# 
+# + contentType - `Content-Type` header value of the received reqeust
+# + topic - WebSubHub `topic` related to the current request
+# + request - Received `http:Request` instance
+# + return - Created `websubhub:UpdateMessage` or `error` if there is any error in execution
+isolated function createUpdateMessage(string contentType, string topic, http:Request request) returns UpdateMessage|error {
+    string|http:HeaderNotFoundError ballerinaPublishEvent = request.getHeader(BALLERINA_PUBLISH_HEADER);
+    if contentType == mime:APPLICATION_FORM_URLENCODED {
+        if ballerinaPublishEvent is string {
+            if ballerinaPublishEvent == "publish" {
+                return {
+                    hubTopic: <string> topic,
+                    msgType: PUBLISH,
+                    contentType: contentType,
+                    content: check retrieveRequestBody(contentType, request)
+                };
+            } else {
+                return {
+                    hubTopic: topic,
+                    msgType: EVENT,
+                    contentType: contentType,
+                    content: ()
+                };
+            }
+        } else {
+            return {
+                hubTopic: topic,
+                msgType: PUBLISH,
+                contentType: contentType,
+                content: check request.getTextPayload()
+            };
+        }
+    } else {
```

Comment:
What if content type is `application/json`. I think in that case there is no way to send an event. I think we don't need to special case this. It would be great if we can just get it to work for any content-type.

Reply:
Here we are invoking `retrieveRequestBody` function to retrieve the payload. It handles all the content types.

Reply:
Updated the logic to disregard content-type when selecting the message-type